odoo/addons/base/models/ir_attachment.py

```diff
@@ -436,8 +438,8 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
                 continue
             # model_attachments = {res_model: {res_id: set(ids)}}
             model_attachments[row['res_model']][row['res_id']].add(row['id'])
-            # Should not retrieve binary fields attachments
-            if row['res_field']:
+            # Should not retrieve binary fields attachments if not explicitly required
+            if not res_field_check and row['res_field']:
```

Comment:
Naming is everything, and this name does not help understanding what's going on.
I suggest the variable name `discard_binary_fields_attachments`, and make it the negation of `res_field_check`.
This will make the code understandable (even more than before!)